pages/blog/api-first-with-asyncapi.md

```diff
@@ -0,0 +1,284 @@
+    - [Info](#info)
+    - [Servers](#servers)
+    - [Channels: Publish / Subscribe](#channels-publish--subscribe)
+    - [Messages](#messages)
+    - [Message Payloads / Schemas](#message-payloads--schemas)
+    - [Reusing Configurations: Operation Traits, Message Traits...](#reusing-configurations-operation-traits-message-traits)
+  - [Different Styles of Event Messages](#different-styles-of-event-messages)
+    - [Notification Messages](#notification-messages)
+    - [State Transfer Messages](#state-transfer-messages)
+    - [Domain Event Messages](#domain-event-messages)
+  - [Next: Java Code Generator for AsyncAPI](#next-java-code-generator-for-asyncapi)
+
+If you are familiar with OpenAPI and OpenAPI Generator API-First workflow:
+
+- First write the OpenAPI definition, collaborating between API providers and API consumers.
+- Then use OpenAPI Generator, either the maven plugin or a CLI, to generate some DTOs and interfaces from your OpenAPI definition.
+- Implementing the generated interfaces you can create a service for the API.
+- As a client, you can use generated the interfaces to consume the API with some HTTP client generated behind the scenes.
+
+When doing API-First with AsyncAPI, the process is similar. After you generate some interfaces and DTOs from your API definition, you use the generated interfaces to produce messages, sending them to the broker, and implement them to consume messages from the broker.
+
+There is still a fundamental difference between OpenAPI and AsyncAPI: OpenAPI is used to document Request-Response / Client-Server APIs, while AsyncAPI is used to document Event-Driven APIs which, except for websockets, are Broker-based. 
+
+And broker-based APIs, unlike Client-Server, are inherently **symmetric**.
+
+## Broker-based APIs are Symmetric
+
+![Client-server vs broker-based EDAs](/img/posts/zenwave/client-server-vs-broker-eda.excalidraw.svg)
+
+Because APIs mediated by a broker are inherently **symmetric** it's difficult to establish the roles of client/server: what represents a `publish` operation from one side will be a `subscribe` operation seen from the other side. Also, a given service can act as a publisher and subscriber on the same API.
+
+For these reasons, to avoid defining the same API operations multiple times from each perspective, we propose to define the API only once from the perspective of the provider of the functionality, which may be a producer, a consumer or both. 
+
+Some definitions:
+
+- SERVICE: An independent piece of software, typically a microservice, that provides a set of capabilities to other services.
+- PROVIDER: The service that implements the functionality of the API. It may be accepting asynchronous command request or publishing business domain events.
+- CLIENT/s: The service/s that makes use of the functionality of the API. It may be requesting asynchronous commands or subscribing to business domain events.
```

Comment:
```suggestion
- CLIENT/s: The service/s that uses the API's functionality. It may be requesting asynchronous commands or subscribing to business domain events.
```